server/channels/app/user.go

```diff
@@ -2212,6 +2224,91 @@ func (a *App) AutocompleteUsersInChannel(rctx request.CTX, teamID string, channe
 	return autocomplete, nil
 }
 
+// autocompleteUsersInChannelWithABAC handles user autocomplete for ABAC-enabled channels
+func (a *App) autocompleteUsersInChannelWithABAC(rctx request.CTX, teamID, channelID, term string, options *model.UserSearchOptions) (*model.UserAutocompleteInChannel, *model.AppError) {
```

Comment:
So anytime this function gets an error, it tries to fetch the autocomplete users from the store. Which is fine, but its doing so in two places. I propose a simplification- don't query store in this function and just return an error. In `AutocompleteUsersInChannel` function above, if this function returned an error, query the store there. This way, we keep the function's responsibility singular.